### ISSUE TITLE
Loop can't be interrupted in vscode, but can in the REPL

### ISSUE THREAD

**Issue by yha** [julia-vscode/julia-vscode#2126]
https://github.com/SciML/DiffEqFlux.jl/issues/539#issuecomment-827966976 shows an example of an infinite loop (due to a bug in `DifferentialEquations`) that can be interrupted (ctrl+c) in the REPL but not in vscode. Repeating it here:
```
using DifferentialEquations

function lotka_volterra!(du, (r,w), (α,β,γ,δ), t)
    du[1] = dr = α*r - β*r*w
    du[2] = dw = γ*r*w - δ*w
end

prob = ODEProblem(lotka_volterra!, [1.0, 1.0], (0.0, 10.0), [100, 700, -200, 1000])
solve(prob)
```

**Comment by pfitzseb** [comment]
Hm, I can interrupt that ok with inline evaluation, but you have to hit Ctrl-C a few times. It's possible that this hits the "force interrupt" heuristic I added a while ago though...

**Comment by yha** [comment]
Hitting ctrl+c many times doesn't help on my machine. This is with Julia 1.5.2. Haven't tried with 1.6.

**Comment by pfitzseb** [comment]
Are you on Windows

**Comment by yha** [comment]
Yes

**Comment by pfitzseb** [comment]
Ah, that explains it then ;) The forced interrupt isn't supported on Windows.

**Comment by yha** [comment]
What does that mean exactly? Usually I can interrupt inline evaluation.

**Comment by pfitzseb** [comment]
That means that the normal mechanism for scheduling an `InterruptException` on the worker process doesn't work in this case (probably because it never `yield`s), but the signal based approach (which is unavailable on Windows) does.

**Comment by gaurav-arya** [comment]
This happens to me on Linux too. See https://github.com/JuliaLang/julia/issues/48154#issuecomment-1373730345

**Comment by gaurav-arya** [comment]
It seems that I simply cannot interrupt any code run with ctrl+enter? E.g. for
```julia
X = 0
for i in 1:10000000000
    X += 1
end
```
when I ctrl+enter the for loop into the REPL, I can't seem interrupt it by ctrl+C or any other means, i.e. I am stuck in this state:
![image](https://user-images.githubusercontent.com/43899491/211140750-15d70c7f-fe57-4c77-baaa-0ed50fb15502.png)

**Comment by j-maffe** [comment]
I'm also getting this issue on a Windows machine. Are there no plans to fix this? It's very annoying since even closing the notebook doesn't interrupt the loop

**Comment by pfitzseb** [comment]
The notebook issue is different from this, I think.

**Comment by j-maffe** [comment]
Should I open a new issue? Do you know if the source of this issue is the extension or the IJulia library?

**Comment by timholy** [comment]
For my own coding, I use vscode for editing (and all the fancy Julia-features y'all have added :heart:) but I execute all my code in the terminal REPL. Some of this personal choice surely comes from my historical habit of delving deep into the guts of Julia and frequently breaking all packages, where using a stripped-down environment for testing things is almost the only possible workflow. But it's also a workflow that I still find myself using now that I'm breaking Julia less often. I recently got a chance to re-examine this choice while introducing a colleague to Julia and vscode, where I found myself failing to explain clearly why I don't use vscode for code execution. So I played with vscode again for a while, running things in its shell, and when I hit this bug I remembered: for me personally, the advantages of getting a quick and reliable response from Ctrl-C seem to typically outweigh the advantages of executing code from within vscode.

Please don't interpret this as a gripe: I'm quite happy with my workflow and love julia-vscode. I just wanted to pass it on because I thought you might be interested in the insight it provides into your user base.

**Comment by pfitzseb** [comment]
I've also gotten very frequently annoyed at this, but properly fixing this would require reworking our code execution mechanism quite a bit -- I don't understand how signals are supposed to be handled in Julia. Generally, you should be able to recover REPL-like interrupts by spamming Ctrl-C for a bit, but you can also easily get unlucky and interrupt some of the extension code at a point we're unlikely to recover from.

**Comment by davidanthoff** [comment]
I (briefly) looked into this. @pfitzseb is the reason we opt-out of the regular sigint handling that we don't want that to happen during the execution of our REPL code, i.e. we want to limit it when user code runs?

Might the following work? We do away with all the custom interrupt handling we have, and then just put our extension code into `disable_sigint` blocks, and all user-code that runs into `reenable_sigint`?

My sense is that this might still lead to odd situations, as apparently Julia base sigint handling is a bit weird in general, but wouldn't this get us at least to the state that one gets in a standard REPL?

**Comment by pfitzseb** [comment]
Yeah, probably. Also there's the standard caveat that none of this will work on Windows anyways (which IIRC was the original motivation for implementing the soft interrupt concept).

**Comment by davidanthoff** [comment]
Ah, I think I remember what the original problem was: we couldn't sent the `SIGINT` signal to the Julia process from the extension on Windows, i.e. `process.kill(pid, 'SIGINT')` doesn't work on Windows.

BUT, I think we can just call `g_terminal.sendText('\x03', false)` and it will actually call the CTRL+C handler in Julia! And presumably also raise the signal on Linux.

**Comment by davidanthoff** [comment]
While I think that https://github.com/julia-vscode/julia-vscode/pull/3758 might get us the same behavior we have in a regular terminal, it probably is blocked by an upstream Julia bug https://github.com/JuliaLang/julia/issues/56977 at the moment.
